kong/plugins/request-transformer/schema/migrations.lua

```diff
@@ -0,0 +1,70 @@
+local next = next
+
+local Migrations = {
+  {
+    name = "2016-03-10-161400_request-transformer-config",
+    up = function(options, dao_factory)
+
+      local plugins, err = dao_factory.plugins:find_by_keys {name = "request-transformer"}
+      if err then
+        return err
+      end
+
+      for _, plugin in ipairs(plugins) do
+        for _, action in ipairs {"remove", "add", "append", "replace"} do
+
+          if plugin.config[action] == nil then
+            plugin.config[action] = {}
+          end
+
+          for _, location in ipairs {"body", "headers", "querystring"} do
+            plugin.config[action][location] = plugin.config[action][location] or {}
+          end
+
+          if plugin.config[action].form ~= nil then
+            plugin.config[action].body = plugin.config[action].form
+            plugin.config[action].form = nil
+          end
+        end
+        local _, err = dao_factory.plugins:update(plugin)
+        if err then
+          return err
+        end
+      end
+    end,
+    down = function(options, dao_factory)
+      local plugins, err = dao_factory.plugins:find_by_keys {name = "request-transformer"}
+      if err then
+        return err
+      end
+
+      for _, plugin in ipairs(plugins) do
+        plugin.config.replace = nil
+        plugin.config.append = nil
+        
+        for _, action in ipairs {"remove", "add"} do
+          for _, location in ipairs {"body", "headers", "querystring"} do
+            if plugin.config[action] ~= nil and next(plugin.config[action][location]) == nil then
```

Comment:
Not all off them, like plugin.config.add.headers can have data and would need to be migrated(basically untouched)